### ISSUE TITLE
[Bug]: No output for dynamic shape

### ISSUE THREAD

**Issue by willdla** [thuml/depyf#63]
### Your current environment

```text
PyTorch version: 2.5.0+cu124

Python version: 3.12.3 (main, Apr 10 2024, 05:33:47) [GCC 13.2.0] (64-bit runtime)
Python platform: Linux-6.8.0-31-generic-x86_64-with-glibc2.39

Versions of relevant libraries:
[pip3] depyf==0.17.0
[pip3] torch==2.5.0
[pip3] torchvision==0.19.1
[conda] Could not collect
```

### 🐛 Describe the bug

```python
import torch

@torch.compile
def function(inputs):
    x = inputs["x"]
    y = inputs["y"]
    x = x.cos().cos()
    if x.mean() > 0.5:
        x = x / 1.1
    return x * y

shape_10_inputs = {"x": torch.randn(10, requires_grad=True), "y": torch.randn(10, requires_grad=True)}
shape_8_inputs = {"x": torch.randn(8, requires_grad=True), "y": torch.randn(8, requires_grad=True)}
# warmup
for i in range(100):
    output = function(shape_10_inputs)
    output = function(shape_8_inputs)

# execution of compiled functions
import depyf
with depyf.prepare_debug("dump_src_dir_dynamic/"):
    output = function(shape_10_inputs)

print(output)
```

if you run the code above you will not get any content in folder `dump_src_dir_dynamic/`

if you comment line 15-17 (the warmup progress) and rerun

you will get the debug information.

notice if you pass TORCH_COMPILE_DEBUG=1 to the script, like:

TORCH_COMPILE_DEBUG=1 python3 test.py

you can successfully get torch_compile_debug folder.

how to get the dynamic shape transform progress using depfy.

Best regards.

**Comment by youkaichao** [comment]
you need to **put the warmup code under `with depyf.prepare_debug`** .

in your case, the compilation finishes after you execute the warmup, and `depyf` does not see any compilation.

**Comment by willdla** [comment]
thanks, this help!
